base/dcp_client_test.go

```diff
@@ -243,8 +238,108 @@ func TestDCPClientMultiFeedConsistency(t *testing.T) {
 	case <-doneChan3:
 		// only vbucket 0 should have rolled back, expect mutation count to be only vbucketZero
 		mutationCount := atomic.LoadUint64(&mutationCount)
-		assert.Equal(t, vbucketZeroExpected, mutationCount)
+		assert.Equal(t, int(vbucketZeroExpected), int(mutationCount))
 	case <-timeout:
 		t.Errorf("timeout waiting for first one-shot feed to complete")
 	}
 }
+
+// TestResumeInterruptedFeed uses persisted metadata to resume the feed
+func TestResumeStoppedFeed(t *testing.T) {
+
+	if UnitTestUrlIsWalrus() {
+		t.Skip("This test only works against Couchbase Server")
+	}
+
+	bucket := GetTestBucket(t)
+	defer bucket.Close()
+
+	var dcpClient *DCPClient
+
+	// create callback
+	mutationCount := uint64(0)
+	counterCallback := func(event sgbucket.FeedEvent) bool {
+		if bytes.HasPrefix(event.Key, []byte(t.Name())) {
+			atomic.AddUint64(&mutationCount, 1)
+			count := atomic.LoadUint64(&mutationCount)
```

Comment:
I think this might be racy (and call Close twice)

You can use the count returned by Add instead of the additional atomic Load.